pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -487,6 +408,278 @@ def phase_str(phi: float) -> str:
+                    target_regions.append([0, targets])
+                    if ch_obj.addressing == "Global":
+                        axes[0].text(
+                            x,
+                            amp_top * 0.98,
+                            "GLOBAL",
+                            fontsize=13,
+                            rotation=90,
+                            ha="left",
+                            va="top",
+                            bbox=q_box,
+                        )
+                    else:
+                        axes[0].text(
+                            x,
+                            tgt_txt_y,
+                            tgt_str,
+                            fontsize=12,
+                            ha="left",
+                            bbox=q_box,
+                        )
+                else:
+                    ti, tf = np.array(coords) / time_scale
+                    target_regions[-1].append(ti)  # Closing previous regions
+                    target_regions.append(
+                        [tf + 1 / time_scale, targets]
+                    )  # New one
+                    for ax in axes:
+                        ax.axvspan(ti, tf, alpha=0.4, color="grey", hatch="//")
+                    axes[0].text(
+                        tf + final_t * 5e-3,
+                        tgt_txt_y,
+                        tgt_str,
+                        ha="left",
+                        fontsize=12,
+                        bbox=q_box,
+                    )
+
+        # Draw the EOM intervals
+        for ch_eom_start_buffer, ch_eom_interval, ch_eom_end_buffer in zip(
+            ch_eom_start_buffers, ch_eom_intervals, ch_eom_end_buffers
+        ):
+            for ax in axes:
+                ch_eom_start_buffer.smooth_draw(ax, decreasing=False)
+                ch_eom_interval.draw(ax)
+                ch_eom_end_buffer.smooth_draw(ax, decreasing=True)
+            tgt_txt_x = ch_eom_start_buffer.ti or ch_eom_interval.ti
+            tgt_txt_y = axes[0].get_ylim()[1]
+            axes[0].text(
+                tgt_txt_x,
+                tgt_txt_y,
+                "EOM",
+                fontsize=12,
+                ha="left",
+                va="top",
+                bbox=eom_box,
+            )
+        # Draw the SLM mask
+        if sampled_seq._slm_mask.targets and sampled_seq._slm_mask.end:
+            tf_m = sampled_seq._slm_mask.end
+            for ax in axes:
+                ax.axvspan(0, tf_m, color="black", alpha=0.1, zorder=-100)
+            tgt_strs = [str(q) for q in sampled_seq._slm_mask.targets]
+            tgt_txt_x = final_t * 0.005
+            tgt_txt_y = axes[-1].get_ylim()[0]
+            tgt_str = "\n".join(tgt_strs)
+            axes[-1].text(
+                tgt_txt_x,
+                tgt_txt_y,
+                tgt_str,
+                fontsize=12,
+                ha="left",
+                bbox=slm_box,
+            )
+
+    return (fig, ch_axes, data)
```

Comment:
Hum, I am not sure we want to return data, but I see why you did it. The structure has to be a bit changed I guess because I think we want the same output as a `draw_sequence`.

I guess this is what you meant by adding a `draw` method to `SequenceSamples`.

Reply:
I think you should have a `draw_sequence`, a `draw_sequence_samples` and a `_draw_channel_content`.

`draw_channel_content` would be the same as the `draw_samples` you have at the moment, and would return (fig, ch_axes, data) (eventually it can be a `draw` method of `DrawChannelContent` ?)

`draw_samples` calls `gather_data` and calls `draw_channel_content` with the `ChannelDrawContent` obtained.

`draw_sequence` does the same thing as `draw_samples`, but add additional features that can only be shown on a `Sequence`.

What do you think ?

Reply:
yes, this encapsulation is a good solution; I'm going for it